src/manager/checkpoint/manager/bottomup.rs

```diff
@@ -0,0 +1,49 @@
+use crate::manager::checkpoint::CheckpointManager;
+use async_trait::async_trait;
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use ipc_gateway::BottomUpCheckpoint;
+use ipc_sdk::subnet_id::SubnetID;
+
+pub struct BottomUpCheckpointManager;
+
+#[async_trait]
+impl CheckpointManager for BottomUpCheckpointManager {
+    type Checkpoint = BottomUpCheckpoint;
+
+    fn parent_subnet(&self) -> &SubnetID {
+        todo!()
+    }
+
+    fn child_subnet(&self) -> &SubnetID {
+        todo!()
+    }
+
+    fn checkpoint_period(&self) -> ChainEpoch {
+        todo!()
+    }
+
+    async fn sync_checkpoint_period(&self) -> anyhow::Result<()> {
```

Comment:
As we mentioned sync, we can probably remove this one and add an asynchronous constructor that gets the checkpoint period? Or maybe `checkpoint_period` could be async and the implementation could try to get the checkpoint period from an attribute from the manager and it if is not set it tries to fetch it from the state of the blockchain (I think I like the latter more)

Reply:
method removed already